### ISSUE TITLE
Lambda Type Stub Generation Missing Parentheses

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1686]
**Describe the bug**
When generating type stubs using pyright --createstub lambdas lose parentheses even at times where it's necessary to be syntactically correct.

**To Reproduce**
Run pyright --createstub on this code,

```python
lambdas = (lambda x: x) or (lambda x: x + 1)
```

The generated stub is,

```python
"""
This type stub file was generated by pyright.
"""

lambdas = lambda x: x or lambda x: x + 1


```

The lack of parentheses for the lambdas makes that stub be syntactically incorrect leading mypy to fail on it.

**Expected behavior**
Type stub generation should pass syntax check.

**VS Code extension or command-line**
I'm using pyright from the command line version 1.1.124.

**Additional context**
The example is based on python code generated from protobuf files and I ran into this trying to generate type stubs for tensorflow. I'd guess other libraries that rely on python protobufs may hit this. Not sure exactly what protobuf options lead to it as trying a minimalistic protobuf file it didn't happen.

**Comment by erictraut** [comment]
Lambdas should not appear in type stubs. They should be replaced by a `Callable` type. I'll look into automating this, but keep in mind that generated stubs almost always require some manual cleanup.

**Comment by erictraut** [comment]
The type stub generator logic generally emits "..." rather than the original expression, but it makes an exception in the case where the variable is at the module level because some of these are type aliases rather than normal variables. I've modified the code to do a better job differentiating between these two cases, so the code will now emit `lambdas = ...` in your example above.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.126, which I just published. It will also be included in the next release of pylance.
